app/Dao/PostDao.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace App\Dao;
+
+use App\Contracts\Dao\PostDaoInterface;
+use App\Http\Requests\CsvUploadRequest;
+use App\Imports\PostsImport;
+use Illuminate\Support\Collection;
+use App\Models\Posts;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class PostDao implements PostDaoInterface
+{
+    /**
+     * Create new post
+     *
+     * @param array $insertData
+     * @return integer
+     */
+    public function insert(array $insertData): int
+    {
+        $post = Posts::create($insertData);
+        return $post->id;
+    }
+
+    /**
+     * Get all posts
+     *
+     * @return Collection
+     */
+    public function getAllPost(): Collection
+    {
+        $user = Auth::user();
+        $posts = Posts::all(); 
+        return $posts;
+        return view('users.detail', compact('user', 'posts'));
+    }
+
+     /**
+     * get public posts
+     *
+     * @return collection
+     */
+    public function getPublicPost(): collection
+    {
+        $userIds = User::select('id')->get()->pluck('id');
+        $posts = Posts::with('comments')->whereIn('created_by', $userIds)->where('public_flag', true)->orderBy('updated_at')->get();
```

Comment:
```suggestion
        $posts = Posts::with('comments')->where('public_flag', true)->orderBy('updated_at')->get();
```
No need to check this here. Every post has created_by  which exist in users table.